File: client/src/features/scene-design/types/scene.ts
import { DesignElement } from '../../../types/product';

// Base GORM fields to match Golang Backend (snake_case)
export interface BaseEntity {
  id: number;
  created_at: string;
  updated_at: string;
}

export interface SceneObject extends Omit<DesignElement, 'type'> {
  type: 'image' | 'text' | 'decor' | 'furniture';
  zIndex: number;
  flipX?: boolean;
}

export interface SceneTemplate extends BaseEntity {
  name: string;
  category: string;
  thumbnail_url: string;
  background_url: string;
  description?: string;
}

export interface InteriorAsset extends BaseEntity {
  name: string;
  url: string;
  category: string;
}

export interface SceneApiResponse<T> {
  message: string;
  data: T;
}

File: server/internal/model/scene.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type SceneTemplate struct {
	ID            uint           `gorm:"primaryKey" json:"id"`
	Name          string         `json:"name"`
	Category      string         `json:"category"`
	ThumbnailURL  string         `json:"thumbnail_url"`
	BackgroundURL string         `json:"background_url"`
	Description   string         `json:"description"`
	CreatedAt     time.Time      `json:"created_at"`
	UpdatedAt     time.Time      `json:"updated_at"`
	DeletedAt     gorm.DeletedAt `gorm:"index" json:"-"`
}

type SceneAsset struct {
	ID        uint           `gorm:"primaryKey" json:"id"`
	Name      string         `json:"name"`
	URL       string         `json:"url"`
	Category  string         `json:"category"` // furniture, decor
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
}
